s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/SourceRecordIterator.java

```diff
@@ -164,6 +164,13 @@ private List<S3SourceRecord> readNext() {
                 return sourceRecords;
             }
 
+            // For bytes transformation, read whole file as 1 record
+            private boolean checkBytesTransformation(final Transformer transformer,
```

Comment:
There should be a specific unit test for this, to ensure it works for the byte array transformers and not the others.